src/bcf-toolkit/Builder/Bcf30/MarkupBuilderExensions.cs

```diff
@@ -1,8 +1,47 @@
+using System.Collections.Generic;
+using BcfToolkit.Model.Bcf30;
+
 namespace BcfToolkit.Builder.Bcf30;
 
-public partial class MarkupBuilder : IMarkupBuilderExtension<MarkupBuilder> {
-  public MarkupBuilder SetServerAssignedId(string id) {
-    _markup.Topic.ServerAssignedId = id;
+public partial class MarkupBuilder {
+  public MarkupBuilder AddHeaderFiles(List<File> files) {
+    files.ForEach(_markup.Header.Files.Add);
+    return this;
+  }
+
+  public MarkupBuilder AddReferenceLinks(List<string> links) {
+    links.ForEach(_markup.Topic.ReferenceLinks.Add);
+    return this;
+  }
+
+  public MarkupBuilder AddLabels(List<string> labels) {
+    labels.ForEach(_markup.Topic.Labels.Add);
+    return this;
+  }
+
+  public MarkupBuilder AddComments(List<Comment> comments) {
+    comments.ForEach(_markup.Topic.Comments.Add);
+    return this;
+  }
+
+  public MarkupBuilder AddViewPoints(List<ViewPoint> viewpoints) {
+    viewpoints.ForEach(_markup.Topic.Viewpoints.Add);
+    return this;
+  }
+
+  public MarkupBuilder AddRelatedTopics(List<string> relatedTopicGuids) {
+    relatedTopicGuids.ForEach(id => {
+      var topic = new TopicRelatedTopicsRelatedTopic {
```

Comment:
Yes because it adds a new `TopicRelatedTopcisRelatedTopic` item to the `RelatedTopics` collection.
`TopicRelatedTopcisRelatedTopic` comes from the bcf schema, but yes, its not really nice :D

Reply:
Damn okay haha